jib-core/src/main/java/com/google/cloud/tools/jib/ncache/LayerEntriesSelector.java

```diff
@@ -92,30 +90,9 @@ public int hashCode() {
     }
   }
 
-  /** Serialized form of a list of {@link LayerEntry}s. */
   @VisibleForTesting
-  static class LayerEntriesTemplate implements ListOfJsonTemplate<LayerEntryTemplate> {
-
-    private final List<LayerEntryTemplate> layerEntryTemplates = new ArrayList<>();
-
-    @Override
-    public List<LayerEntryTemplate> getList() {
-      return layerEntryTemplates;
-    }
-
-    /**
-     * Serializes a list of {@link LayerEntry}s into a new {@link LayerEntriesTemplate}. The list is
-     * sorted by source file first, then extraction path (see {@link LayerEntryTemplate#compareTo}).
-     *
-     * @param layerEntries the list of {@link LayerEntry}s
-     */
-    @VisibleForTesting
-    LayerEntriesTemplate(ImmutableList<LayerEntry> layerEntries) {
-      for (LayerEntry layerEntry : layerEntries) {
-        layerEntryTemplates.add(new LayerEntryTemplate(layerEntry));
-      }
-      Collections.sort(layerEntryTemplates);
-    }
+  static List<LayerEntryTemplate> toSortedJsonTemplates(List<LayerEntry> layerEntries) {
+    return layerEntries.stream().map(LayerEntryTemplate::new).sorted().collect(Collectors.toList());
```

Comment:
I think we might want to keep it non-stream style since we might not want to incur the stream overhead in this class as it one of the lower-level methods.